vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java

```diff
@@ -281,15 +283,69 @@ public void iGoTo(String relativeURL)
      */
     @When("I close the current window")
     public void closeCurrentWindow()
+    {
+        tryToCloseCurrentWindow(false);
+    }
+
+    /**
+     * Trying to close <b>current window</b> with JavaScript method 'close()'.
+     * If an alert window via 'onbeforeunload' event opens, it must be checked and handled in the next steps.
+     * If an alert window does not open, closes current window and switches to the window from which redirected
+     * to current window.
+     * <p>
+     * Each browser <b>window</b> or <b>tab</b> is considered to be a separate <b>window object</b>. This object holds
+     * corresponding <b>Document</b> object, which itself is a html page. So this method applies to both windows and
+     * tabs.
+     * <p>
+     * Actions performed at this step:
+     * <ul>
+     * <li>Receives all opened browser windows
+     * <li>Identifies current window and try to close it
+     * <li>If an alert window via 'onbeforeunload' event opens, step completes execution
+     * (alert is expected to be handled in next steps)
+     * <li>If an alert window does not open, closes current window and
+     *  switches back to the window from which rederection to current window was performed
+     * </ul>
+     * <p>
+     * Note that this step can only be used if the current window was opened using the step
+     * 'When I open URL `$pageUrl` in new window'.
+     * <p>
+     * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
+     * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
+     * @see <a href="https://www.w3schools.com/jsref/event_onbeforeunload.asp"><i>Event 'onbeforeunload'</i></a>
+     * @see <a href="https://www.w3schools.com/js/js_popup.asp"><i>JavaScript Popup Boxes</i></a>
+     */
+    @When("I try to close the current window with alerts handling")
+    public void closeCurrentWindowWithAlertsHandling()
+    {
+        tryToCloseCurrentWindow(true);
+    }
+
+    private void tryToCloseCurrentWindow(boolean isAlertHandling)
     {
         WebDriver driver = getWebDriver();
         String currentWindow = driver.getWindowHandle();
         for (String window : driver.getWindowHandles())
         {
             if (!window.equals(currentWindow))
             {
-                driver.close();
-                driver.switchTo().window(window);
+                if (isAlertHandling)
```

Comment:
I propose to separate methods with/without alert handling and remove this condition. We always know if alert handling is required, so why not use the proper method

Reply:
> I propose to separate methods with/without alert handling and remove this condition. We always know if alert handling is required, so why not use the proper method

The main goal of creating the common method was to get rid of code duplication. If we will put current common logic in different methods, we will have to duplicate the code or move the common logic into several more methods. Or I misunderstand you?